src/BlazorWebView/src/Maui/Android/WebKitWebViewClient.cs

```diff
@@ -65,7 +87,60 @@ public override bool ShouldOverrideUrlLoading(AWebView? view, IWebResourceReques
 				}
 			}
 
-			return base.ShouldOverrideUrlLoading(view, request);
+			return false;
+		}
+
+		private void TryStartActivityFromUri(string? uriString, bool uriHasIntentScheme)
+		{
+			if (_webViewHandler is null || uriString is null)
+			{
+				return;
+			}
+
+			try
+			{
+				var intent = Intent.ParseUri(uriString, IntentUriType.Scheme);
+
+				if (intent is not null)
+				{
+					_webViewHandler.Context.StartActivity(intent);
+				}
+			}
+			catch (URISyntaxException)
+			{
+				// No-op. The URI was malformed.
+				return;
+			}
+			catch (ActivityNotFoundException)
+			{
+				if (!uriHasIntentScheme)
+				{
+					// No-op. This behavior is consistent with other platforms when a URI with an unknown scheme is clicked.
+					return;
+				}
+
+				// Chrome for Android allows specifying a fallback URL in case an intent cannot be resolved.
+				// Since the Android WebView cannot handle "intent"-schemed URIs, we add our own support for the fallback
+				// URL here.
+				// See: https://developer.chrome.com/docs/multidevice/android/intents/
```

Comment:
TBH the sophistication of the logic here is mildly concerning. It will be expensive to keep testing and maintaining this in an ongoing way. We'd need quite a lot of different CTI test flows to exercise all this. Is there much of a chance that people will come up with other sub-cases and ask for other variations in behavior?

Since Google didn't see fit to endow Android WebView with the ability to handle this by default, I wonder if it's really needed for BlazorWebView to do so.

Might it be sufficient to extend the `ExternalNavigationStarting` callback to cover the non-HTTP(S)-URL case too, and then developers can put in whatever logic they want based on the specific URL types they intent to use in their app?

I totally accept I might be missing some important point here. As a non-Android user I have limited intuition for how much intent links are used on that OS and what range of support is needed for a core developer experience. Please let me know if you think I'm failing to grasp how right and important this feature is!